subworkflows/local/bin_qc.nf

```diff
@@ -0,0 +1,182 @@
+        /*
+         * BUSCO
+         */
+        if (!ch_busco_db.isEmpty()) {
+            if (ch_busco_db.extension in ['gz', 'tgz']) {
+                // Expects to be tar.gz!
+                BUSCO_DB_PREPARATION(ch_busco_db)
+                ch_db_for_busco = BUSCO_DB_PREPARATION.out.db.map { meta, db ->
+                    [[id: meta, lineage: 'Y'], db]
+                }
+            }
+            else if (ch_busco_db.isDirectory()) {
+                // Set meta to match expected channel cardinality for BUSCO
+                ch_db_for_busco = Channel
+                    .of(ch_busco_db)
+                    .collect { db ->
+                        def basename = db.getBaseName()
+                        def lineage = basename.contains('odb10') ? 'Y' : 'N'
+                        [[id: basename, lineage: lineage], db]
+                    }
+            }
+        }
+        else {
+            // Set BUSCO database to empty to allow for --auto-lineage
+            ch_db_for_busco = Channel
+                .of([[lineage: ''], []])
+                .collect()
+        }
+
+        if (params.save_busco_db) {
+            // publish files downloaded by Busco
+            ch_downloads = BUSCO.out.busco_downloads
+                .groupTuple()
+                .map { _lin, downloads -> downloads[0] }
+                .toSortedList()
+                .flatten()
+            BUSCO_SAVE_DOWNLOAD(ch_downloads)
+        }
+
+        BUSCO(ch_input_bins_for_qc, ch_db_for_busco)
+
+        BUSCO_SUMMARY(
+            BUSCO.out.summary_domain.collect { v -> v[1] }.ifEmpty([]),
+            BUSCO.out.summary_specific.collect { v -> v[1] }.ifEmpty([]),
+            BUSCO.out.failed_bin.collect { v -> v[1] }.ifEmpty([])
+        )
+
+        ch_multiqc_files = ch_multiqc_files.mix(
+            BUSCO.out.summary_domain.mix(BUSCO.out.summary_specific).map { it[1] }
+        )
+        qc_summary = BUSCO_SUMMARY.out.summary
+        ch_versions = ch_versions.mix(BUSCO.out.versions.first())
+    }
+    else if (params.binqc_tool == "checkm") {
+        /*
+         * CheckM
+         */
+        ch_bins_for_checkmlineagewf = ch_input_bins_for_qc
+            .groupTuple()
+            .filter { meta, _bins ->
+                meta.domain != "eukarya"
+            }
+            .multiMap { meta, fa ->
+                reads: [meta, fa]
+                ext: fa.extension.unique().join("")
+            }
+
+        CHECKM_LINEAGEWF(ch_bins_for_checkmlineagewf.reads, ch_bins_for_checkmlineagewf.ext, ch_checkm_db)
+        ch_versions = ch_versions.mix(CHECKM_LINEAGEWF.out.versions.first())
+
+        ch_checkmqa_input = CHECKM_LINEAGEWF.out.checkm_output
+            .join(CHECKM_LINEAGEWF.out.marker_file)
+            .map { meta, dir, marker ->
+                [meta, dir, marker, []]
+            }
+
+        CHECKM_QA(ch_checkmqa_input, [])
+
+        COMBINE_BINQC_TSV(CHECKM_QA.out.output.collect { v -> v[1] })
```

Comment:
same again, what type of file is the 'v' referring to? I like to be ultra verbose as we have quite diverse developers contributing to the same pipeline :)

Reply:
TBH, I was using `v` because the Nextflow documentation no longer encourages the use of `it`, and instead uses `v` in the examples:: https://www.nextflow.io/docs/stable/script.html#closures. But yeah, you're absolutely right, it's a very lazy thing to do.